x-pack/plugins/task_manager/server/task.ts

```diff
@@ -83,58 +85,62 @@ export interface FailedTaskResult {
   status: TaskStatus.Failed;
 }
 
-export const validateRunResult = Joi.object({
-  runAt: Joi.date().optional(),
-  schedule: Joi.object().optional(),
-  error: Joi.object().optional(),
-  state: Joi.object().optional(),
-}).optional();
-
```

Comment:
This was added back when TS wasn't much adopted, but we can now rely on Typescript for this.
For a task to return an invalid response they would have to lie to the compiler... we're comfortable with relying on compile time validation of Task Type Definitions at this point.